server/routes/sources/sources-router.js

```diff
@@ -2,53 +2,84 @@ import express from 'express';
 import AppError from '../../errors/AppError.js';
 import {
   findSourceCountry,
-  updateSourceById,
+  getSourceByIdSourceName,
+  getCrosswalkByIdSourceName,
+  updateSourceByIdSourceName,
   createSource,
   createCrosswalk,
   getAllSources,
+  updateCrosswalkByIdSourceName,
 } from './sources-queries.js';
+import validateSource from './sources-validations.js';
 
 const sourcesRouter = express.Router();
 
 sourcesRouter.get('/', async (req, res) => {
-  const { id, country } = req.query;
+  const { country, idSourceName, sources } = req.query;
 
-  const idSource = !id ? '*' : id;
-  const source = !country
-    ? await getAllSources()
-    : await findSourceCountry(country, idSource);
-  res.status(200).json(source ?? {});
+  if (!idSourceName && sources === 'All') {
+    const foundSources = await getAllSources();
+    if (!foundSources || foundSources.length === 0)
+      throw new AppError(400, 'Error getting source');
+    res.status(200).json(foundSources ?? {});
+  }
+
+  if (idSourceName) {
+    const responseSource = await getSourceByIdSourceName(idSourceName);
+    const responseCrosswalk = await getCrosswalkByIdSourceName(idSourceName);
+    if (!responseSource || responseSource.length === 0)
+      throw new AppError(400, 'Error getting source');
+    res
+      .status(200)
+      .json({ source: responseSource, crosswalk: responseCrosswalk } ?? {});
+  }
+
+  if (country) {
+    const responseSourceCountry = await findSourceCountry(country);
+    if (!responseSourceCountry || responseSourceCountry.length === 0)
+      throw new AppError(400, 'Error getting source');
+    res.status(200).json(responseSourceCountry ?? {});
+  }
 });
```

Comment:
Can we get here? If so, what should be returned? I assume that would mean bad inputs.